CRM/Contribute/Selector/Search.php

```diff
@@ -402,6 +402,13 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
         }
       }
 
+      $row['balance_due'] = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(
+        $result->contribution_id,
```

Comment:
OK - it would actually be easy to do in an extension - ie. add an api in the extension & just add a region in the tpl & call crmAPI from there